eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs

```diff
@@ -135,6 +138,14 @@ import GHC.TypeLits
 import NoThunks.Class (InspectHeapNamed (..), NoThunks)
 import Prelude hiding (lookup)
 
+data Addr28Extra
+  = Addr28Extra
+      {-# UNPACK #-} !Word64 -- Payment Addr
+      {-# UNPACK #-} !Word64 -- Payment Addr
+      {-# UNPACK #-} !Word64 -- Payment Addr
+      {-# UNPACK #-} !Word64 -- Payment Addr (32bits) + ... +  0/1 for Testnet/Mainnet + 0/1 Script/Pubkey
+  deriving (Eq)
+
 data TxOut era
```

Comment:
BTW, the Eq instance below is incorrect now that we export all constructors (i.e. we don't enforce a canonical form), correct? Though this is probably out of scope for this PR.

Reply:
Quite the opposite. This PR enforces just that by introducing this `Addr28Extra` data type that does not have the constructor exported, which means it can be only created by the `encodeAddress28` function, which is canonical.